components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3692,5 +3249,628 @@ subroutine get_time_space_phys_variables( &
 
 end subroutine get_time_space_phys_variables
 
+subroutine cloud_sedimentation(kts,kte,    &
+   qc_incld,rho,inv_rho,lcldm,acn,inv_dzq,&
+   dt,odt,dnu,lammin,lammax,log_predictNc, & 
+   qc, nc, nc_incld,mu_c,nu,lamc,prt_liq,qc_tend,nc_tend) 
+   
+   implicit none 
+   integer, intent(in) :: kts, kte
+
+   real(rtype), intent(in), dimension(kts:kte) :: qc_incld
+
+   real(rtype), intent(in), dimension(kts:kte) :: rho
+   real(rtype), intent(in), dimension(kts:kte) :: inv_rho 
+   real(rtype), intent(in), dimension(kts:kte) :: lcldm 
+   real(rtype), intent(in), dimension(kts:kte) :: acn 
+   real(rtype), intent(in), dimension(kts:kte) :: inv_dzq
+   real(rtype), intent(in) :: dt 
+   real(rtype), intent(in) :: odt 
+   real(rtype), dimension(:), intent(in) :: dnu
+   real(rtype), intent(in) :: lammin
+   real(rtype), intent(in) :: lammax 
+   logical, intent(in) :: log_predictNc 
+
+   real(rtype), intent(inout), dimension(kts:kte) :: qc 
+   real(rtype), intent(inout), dimension(kts:kte) :: nc 
+   real(rtype), intent(inout), dimension(kts:kte) :: nc_incld 
+   real(rtype), intent(inout), dimension(kts:kte) :: mu_c 
+   real(rtype), intent(inout), dimension(kts:kte) :: nu
+   real(rtype), intent(inout), dimension(kts:kte) :: lamc
+   real(rtype), intent(inout) :: prt_liq
+   real(rtype), intent(inout), dimension(kts:kte) :: qc_tend 
+   real(rtype), intent(inout), dimension(kts:kte) :: nc_tend 
+   
+   logical :: log_qxpresent, qc_present, two_moment
+   integer :: k 
+   integer :: kbot, ktop, kdir, k_qxtop, k_qxbot, k_temp 
+   integer :: tmpint1 
+
+   real(rtype) :: dt_left 
+   real(rtype) :: prt_accum 
+   real(rtype) :: Co_max 
+   real(rtype) :: dt_sub
+   real(rtype), dimension(kts:kte) :: V_qc
+   real(rtype), dimension(kts:kte) :: V_nc 
+   real(rtype), dimension(kts:kte) :: flux_qx
+   real(rtype), dimension(kts:kte) :: flux_nx
+   real(rtype) :: fluxdiv_qx, fluxdiv_nx 
+ 
+   real(rtype) :: tmp1, tmp2, dum 
+
+   ktop = kts
```

Comment:
The whole distinction between `ktop` and `kts` is a bit bothersome.  I understand that the idea is the first column indice, `kts` may not necessarily correspond with either the top or bottom (depending on model).  So we want flexibility.  But everywhere we make the distinction we just hard-code anyway.  This tells me that we we may be able to get rid of one of these sets of variables.  Maybe keeping ktop and kbot because they make physical sense, like for sedimentation.  Any thoughts?

Reply:
I'm going to resolve this comment stream and refer to my later comment on `ktop` and `kbot`